joomla/joomla-cms
# Clean up old code in cache.php file

## Summary of Changes
- Remove old php4 style to catch exception which currently is useless. 
- Add shorten version of ternary operators.
- In `lock` and `unlock` methods direct call to handler.
### Testing Instructions

Code review or real test:
1. Get latest staging Joomla.
2. Enable cache and set timeout to low value, for ex. 1 minute.
3. Go to frontend and check if pages work as before.
4. Go to backend and some cache items should be visible, clear only expired cache, clear all cache.
